rclcpp_components/test/test_component_manager_api.cpp

```diff
@@ -211,6 +211,71 @@ TEST_F(TestComponentManager, components_api)
     EXPECT_EQ(result->unique_id, 0u);
   }
 
+  {
+    // use_dedicated_thread
+    auto request = std::make_shared<composition_interfaces::srv::LoadNode::Request>();
+    request->package_name = "rclcpp_components";
+    request->plugin_name = "test_rclcpp_components::TestComponentFoo";
+    request->node_name = "test_component_dedicated_thread";
+    rclcpp::Parameter use_dedicated_thread("use_dedicated_thread",
+      rclcpp::ParameterValue(true));
+    request->extra_arguments.push_back(use_dedicated_thread.to_parameter_msg());
+
+    auto future = composition_client->async_send_request(request);
+    auto ret = exec->spin_until_future_complete(future, 5s);  // Wait for the result.
+    auto result = future.get();
+    EXPECT_EQ(ret, rclcpp::FutureReturnCode::SUCCESS);
+    EXPECT_EQ(result->success, true);
+    EXPECT_EQ(result->error_message, "");
+    std::cout << result->full_node_name << std::endl;
+    EXPECT_EQ(result->full_node_name, "/test_component_dedicated_thread");
+    EXPECT_EQ(result->unique_id, 7u);
+  }
+
+  {
+    // use_dedicated_thread (used for testing unload_node)
+    auto request = std::make_shared<composition_interfaces::srv::LoadNode::Request>();
+    request->package_name = "rclcpp_components";
+    request->plugin_name = "test_rclcpp_components::TestComponentFoo";
+    request->node_name = "test_component_dedicated_thread2";
+    rclcpp::Parameter use_dedicated_thread("use_dedicated_thread",
+      rclcpp::ParameterValue(true));
+    request->extra_arguments.push_back(use_dedicated_thread.to_parameter_msg());
+
+    auto future = composition_client->async_send_request(request);
+    auto ret = exec->spin_until_future_complete(future, 5s);  // Wait for the result.
+    auto result = future.get();
+    EXPECT_EQ(ret, rclcpp::FutureReturnCode::SUCCESS);
+    EXPECT_EQ(result->success, true);
+    EXPECT_EQ(result->error_message, "");
+    std::cout << result->full_node_name << std::endl;
```

Comment:
nitpick: is this needed?

Reply:
the way of unloading nodes which use dedicated thread is different of normal nodes, so i added this test code, what do you think about  it? @fujitatomoya .if we needn't it , i can remove it.
https://github.com/ros2/rclcpp/blob/0716a84f32bca9535cc690ae72bf78433ea61a78/rclcpp_components/src/component_manager.cpp#L282-L293

Reply:
i mean test case is needed but `std::cout << result->full_node_name << std::endl;`? sorry i wasnt clear before.